src/shell-integration/bash/ghostty.bash

```diff
@@ -40,35 +39,40 @@ if [ -n "$GHOSTTY_BASH_INJECT" ]; then
     builtin unset GHOSTTY_BASH_UNEXPORT_HISTFILE
   fi
 
-  # Manually source the startup files, respecting the injected flags like
-  # --norc and --noprofile that we parsed with the shell integration code.
-  #
-  # See also: run_startup_files() in shell.c in the Bash source code
-  if builtin shopt -q login_shell; then
-    if [[ $ghostty_bash_inject != *"--noprofile"* ]]; then
-      [ -r /etc/profile ] && builtin source "/etc/profile"
-      for rcfile in "$HOME/.bash_profile" "$HOME/.bash_login" "$HOME/.profile"; do
-        [ -r "$rcfile" ] && { builtin source "$rcfile"; break; }
-      done
-    fi
-  else
-    if [[ $ghostty_bash_inject != *"--norc"* ]]; then
-      # The location of the system bashrc is determined at bash build
-      # time via -DSYS_BASHRC and can therefore vary across distros:
-      #  Arch, Debian, Ubuntu use /etc/bash.bashrc
-      #  Fedora uses /etc/bashrc sourced from ~/.bashrc instead of SYS_BASHRC
-      #  Void Linux uses /etc/bash/bashrc
-      #  Nixos uses /etc/bashrc
-      for rcfile in /etc/bash.bashrc /etc/bash/bashrc /etc/bashrc; do
-        [ -r "$rcfile" ] && { builtin source "$rcfile"; break; }
-      done
-      if [[ -z "$GHOSTTY_BASH_RCFILE" ]]; then GHOSTTY_BASH_RCFILE="$HOME/.bashrc"; fi
-      [ -r "$GHOSTTY_BASH_RCFILE" ] && builtin source "$GHOSTTY_BASH_RCFILE"
+  # Manually source the startup files. See INVOCATION in bash(1) and
+  # run_startup_files() in shell.c in the Bash source code.
+  function __ghostty_bash_startup() {
+    builtin local rcfile
+
+    if builtin shopt -q login_shell; then
+      if [[ $__ghostty_bash_flags != *"--noprofile"* ]]; then
+        [ -r /etc/profile ] && builtin source "/etc/profile"
+        for rcfile in ~/.bash_profile ~/.bash_login ~/.profile; do
+          [ -r "$rcfile" ] && { builtin source "$rcfile"; break; }
+        done
+      fi
+    else
+      if [[ $__ghostty_bash_flags != *"--norc"* ]]; then
+        # The location of the system bashrc is determined at bash build
+        # time via -DSYS_BASHRC and can therefore vary across distros:
+        #  Arch, Debian, Ubuntu use /etc/bash.bashrc
+        #  Fedora uses /etc/bashrc sourced from ~/.bashrc instead of SYS_BASHRC
+        #  Void Linux uses /etc/bash/bashrc
+        #  Nixos uses /etc/bashrc
+        for rcfile in /etc/bash.bashrc /etc/bash/bashrc /etc/bashrc; do
+          [ -r "$rcfile" ] && { builtin source "$rcfile"; break; }
+        done
+        if [[ -z "$GHOSTTY_BASH_RCFILE" ]]; then GHOSTTY_BASH_RCFILE=~/.bashrc; fi
```

Comment:
I have doubts about the expansion of the tilde here. Bash does expand ~ in the middle, but pure sh does not:
```console
[zbyszek@x1c ghostty]$ sh -c 'echo VAR=~'
VAR=~
[zbyszek@x1c ghostty]$ bash -c 'echo VAR=~'
VAR=/home/zbyszek
```
It'd be good to verify that it is expanded as expected.